### ISSUE TITLE
Build failure: VirtualBox with kvm patch

### ISSUE THREAD

**Issue by Quackification** [NixOS/nixpkgs#369311]
## Steps To Reproduce

1. enable virtualbox with kvm in configuration.nix
2. rebuild switch

## Build log

<details>

<summary>Build Log</summary>

```

trying https://github.com/cyberus-technology/virtualbox-kvm/releases/download/dev-20240828/kvm-backend-7.1.4-dev-20240828.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M  0     0    0    >
curl: (22) The requested URL returned error: 404
error: cannot download virtualbox-7.1.4-kvm-dev-20240828.patch from any mirror
```

</details>

[## Additional context]
(https://github.com/cyberus-technology/virtualbox-kvm/releases/download/dev-20240828/kvm-backend-7.1.4-dev-20240828.patch) does not exist

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.6, NixOS, 24.11 (Vicuna), 24.11.20241227.d49da4c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"home-manager-24.11.tar.gz, nixos-24.11"`
 - nixpkgs: `/nix/store/salp9r9j3pj9cwqf06wchs16hy8g882k-source`

## Notify maintainers

@svanderburg @blitz @FriedrichAltheide 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Quackification** [comment]
The one that actually exists is called [https://github.com/cyberus-technology/virtualbox-kvm/releases/download/dev-20241220/kvm-backend-7.1.4-dev-20241220.patch](https://github.com/cyberus-technology/virtualbox-kvm/releases/download/dev-20241220/kvm-backend-7.1.4-dev-20241220.patch). probably mixed up the urls because [https://github.com/cyberus-technology/virtualbox-kvm/releases/download/dev-20240828/kvm-backend-7.0.20-dev-20240828.patch](https://github.com/cyberus-technology/virtualbox-kvm/releases/download/dev-20240828/kvm-backend-7.0.20-dev-20240828.patch) also exists.

**Comment by Quackification** [comment]
90% sure the fix is just to change kvmPatchHash = "sha256-SYyD79iN6Sp/Mxat+ml3fee9X1vFUFyrwHPnQNboc1c="; to the new sha256, but what do i know

**Comment by Quackification** [comment]
found the offending code
```
++ optional enableKvm (
      let
        patchVboxVersion =
          # There is no updated patch for 7.0.22 yet, but the older one still applies.
          if finalAttrs.virtualboxVersion == "7.0.22" then "7.0.20" else finalAttrs.virtualboxVersion;
      in
      fetchpatch {
        name = "virtualbox-${finalAttrs.virtualboxVersion}-kvm-dev-${finalAttrs.kvmPatchVersion}.patch";
        url = "https://github.com/cyberus-technology/virtualbox-kvm/releases/download/dev-${finalAttrs.kvmPatchVersion}/kvm-backend-${patchVboxVersion}-dev-${finalAttrs.kvmPatchVersion}.patch";
        hash = finalAttrs.kvmPatchHash;
      }
    )
```

**Comment by FriedrichAltheide** [comment]
Should be fixed once https://github.com/NixOS/nixpkgs/pull/368897 landed in nixos-24.11 (https://nixpk.gs/pr-tracker.html?pr=368897)
